test/local_test.go

```diff
@@ -0,0 +1,69 @@
+package test
+
+import (
+	"errors"
+	"gopkg.in/yaml.v3"
+	"io"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+type parsedCRD struct {
+	APIVersion string `yaml:"apiVersion"`
+	Kind       string `yaml:"kind"`
+	Metadata   struct {
+		Annotations struct {
+			EksAmazonawsComRoleArn              string `yaml:"eks.amazonaws.com/role-arn"`
+			EksAmazonawsComStsRegionalEndpoints string `yaml:"eks.amazonaws.com/sts-regional-endpoints"`
+		} `yaml:"annotations"`
+		Name string `yaml:"name"`
+	} `yaml:"metadata"`
+}
+
+func TestServiceAccountsUseStsRegionalEndpoint(t *testing.T) {
+	addonsPath := "../modules/cluster/addons/"
+	addonFiles, err := ioutil.ReadDir(addonsPath)
```

Comment:
`ioutil` package is [deprecated in Go 1.16](https://pkg.go.dev/io/ioutil#ReadDir). I know you specified Go 1.15 in CI but I don't think we have reasons to use the old version.

```suggestion
	addonFiles, err := os.ReadDir(addonsPath)
```

Reply:
The project was using 1.15 before https://github.com/cookpad/terraform-aws-eks/blob/main/test/go.mod#L3
Happy to upgrade to 1.17 👍